src/core/thread/meshcop_dataset.cpp

```diff
@@ -356,14 +354,32 @@ ThreadError Dataset::Set(const otOperationalDataset &aDataset, bool aActive)
     return error;
 }
 
-const Timestamp &Dataset::GetTimestamp(void) const
+const Timestamp *Dataset::GetTimestamp(void) const
 {
-    return mTimestamp;
+    const uint8_t *tlv = reinterpret_cast<const uint8_t *>(Get(static_cast<Tlv::Type>(mType)));
```

Comment:
Rather than casting this to a `uint8_t`, I think it's better if we cast to the proper `ActiveTimestampTlv` and `PendingTimestampTlv` depending on the type.